spec/job_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+require 'sequenceserver/job'
+require 'sequenceserver/blast/job'
+require 'rack/test'
+
+# Test Job class from module SequenceServer
+module SequenceServer
+  describe 'blank Job' do
+    my_job = Job.new
+
+    it 'should have an id' do
+      regex = /\A(urn:uuid:)?[\da-f]{8}-([\da-f]{4}-){3}[\da-f]{12}\z/i
+      expect(my_job.id).to match(regex)
+    end
+
+    it 'should have a time of submission' do
+      expect(my_job.submitted_at).to be_kind_of(Time)
+    end
+
+    it 'should have an false exitstatus' do
+      expect(my_job.done?).to be_falsey
+    end
+
+    it 'should not be completed yet' do
+      expect(my_job.completed_at).to be(nil)
+    end
+
+    it 'should have a job directory' do
+      expect("#{__dir__}/dotdir/#{my_job.id}").not_to be_empty
+    end
+  end
+
+  describe 'BLAST Job' do
+    # Test Job class from module Blast and params
+    ENV['RACK_ENV'] = 'test'
+    include Rack::Test::Methods
+
+    before do
+      SequenceServer.init(database_dir: "#{__dir__}/database/v5/sample")
+
+      @params = {
+        sequence: ">seq1\nAGCTAGCTAGCT\n>seq2\nAATAGCTA",
```

Comment:
I suggest adding several more scenarios. 

E>.g, including spaces etc. (you're not testing the strip functionality)

Reply:
Ok that's doable.
What do you think of 
`@params = {
        sequence: ">seq1   \n AGCTAGCTAGCT \n>  SI2.2.0_80257 Si_gnF.scaffold" \
        "[2092364..2095011].pep_1  \n  AATAGCTA\n>s7:S11.0.9.1233 Fi_gnG.scaf" \
        "[5567] \n    PPKKLLMMOOPPPTTRYUIQWERTZ     ",`

Description, protein, and strip are included.

If whitespaces at the beginning of the fasta-header should be included, then the `calculate_query_length` function must be adjusted. The same goes for whitespaces in between the sequence. By now only leading and trailing whitespaces in a sequence are deleted.

I think we should leave '>' as the start of every fasta and additionally there shouldn't be a need of deleting whitespaces inside sequences as normal fasta files should not contain whitespace inside sequences. What do you think of that?